src/main/java/org/torproject/onionoo/updater/RdnsLookupRequest.java

```diff
@@ -23,7 +27,6 @@
   public RdnsLookupRequest(
       ReverseDomainNameResolver reverseDomainNameResolver,
       RdnsLookupWorker parent, String address) {
-    this.reverseDomainNameResolver = reverseDomainNameResolver;
```

Comment:
I wonder, should we simply pass a `DomainNameSystem` reference here instead? Right now it's implemented as a singleton, right? That is hard to test then.

Somewhat unrelated, the class name `DomainNameSystem` is a bit ambiguous. Can't we keep the `ReverseDomainNameResolver` and just rewrite the contained code? At least that class name says what the class does/is responsible for.

Reply:
`ReverseDomainNameResolver` is not really the resolver, but just a part of the updater process that manages requests and workers. `DomainNameSystem` is currently implemented with a static method.

Reply:
This is the only remaining concern I have: does `DomainNameSystem` have to be implemented with a static method? This effectively makes the class a singleton. Do you mind changing the static block to a constructor and the static method to an instance method? Otherwise this may bite us in the future.